src/formatDate.js

```diff
@@ -8,7 +8,48 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [, , , fromSeparator] = fromFormat;
+  const [, , , toSeparator] = toFormat;
+
+  const dateArray = date.split(fromSeparator);
+  const modifiedDate = [];
+
+  const fromFormatLength = fromFormat.join('').length;
+  const toFormatLength = toFormat.join('').length;
+
+  if (fromFormatLength > toFormatLength) {
+    const fromYearIndex = fromFormat.indexOf('YYYY');
+    const splitArrayDate = dateArray[fromYearIndex].split('');
+
+    for (let i = 1; i < 3; i++) {
+      splitArrayDate.shift();
+    }
+
+    dateArray[fromYearIndex] = splitArrayDate.join('');
+
+    fromFormat[fromYearIndex] = 'YY';
+  } else if (fromFormatLength < toFormatLength) {
+    const fromYearIndex = fromFormat.indexOf('YY');
+    const splitArrayDate = dateArray[fromYearIndex].split('');
+
+    if (dateArray[fromYearIndex] < 30) {
+      splitArrayDate.unshift(20);
+    } else {
+      splitArrayDate.unshift(19);
```

Comment:
The logic for determining the century (19 or 20) based on the year being less than 30 is a common practice but might not be suitable for all use cases. Consider making this logic configurable or providing a more robust solution.